orttraining/orttraining/python/experimental/model_desc_validation.py

```diff
@@ -184,15 +184,22 @@ def _add_input_description(self, node, name, shape, dtype=None, attr_name=None):
             shape (list): shape of input description
             dtype (torch.dtype): input data type
             attr_name (str, default is None): friendly name to allow direct access to the output description
+            ignore_duplicate (bool, default is False): silently skips addition of duplicate inputs
         '''
 
         assert isinstance(name, str) and len(name) > 0, "'name' is an invalid input name"
+        not_found = True
+        if ignore_duplicate is False:
```

Comment:
`is False` is dangerous here and later, because it compares to specific object `False`, and, for example, `1 is True` is False. Probably better to re-write as  `if not ignore_duplicate:`